dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CachingFileTree.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.helpers;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implements a FileTree that wraps a specified FileTree, delegates to
+ * the underlying FileTree's methods, saves the returned values, and
+ * returns the saved value when a method is invoked multiple times with
+ * the same arguments.
+ *
+ * For example, the first call to `readFile("/foo.txt")` is delegated
+ * to the underlying FileTree, and the resulting content is saved and
+ * returned.  For subsequent calls to `readFile("/foo.txt")`, the saved
+ * content is returned, and the underlying FileTree is not used.
+ *
+ * Useful for avoiding repeated accesses to a FileTree-abstracted
+ * resource when the code makes multiple passes over the same files,
+ * without needing to explicitly propagate the retrieved information
+ * (by passing it down the call stack, saving it in a variable, etc).
+ */
+public class CachingFileTree implements FileTree {
+
+    private final FileTree fileTree;
+    private final Map<String, String> pathToContent = new HashMap<>();
```

Comment:
Code comment, this is a map between the file paths and content  whereas the next line maps out what files exist in which directories.